### PR TITLE
Add +200 colors for Sharezone Plus users

### LINKED ISSUE #854: Add color picker to pick +200 colors for Sharezone Plus users
Sharezone Plus should be able to select a color from a pool of +200 colors. We can build a color picker like this: First dialog, choose the base color (like `Colors.blue`). When you press on the base color, you get all blue options (like `Colors.blue[100]`, `Colors.blue[200]` ... `Colors.blue[900]`).

### LINKED ISSUE #855: Add note that a user can unlock +200 additional colors when upgrading to Sharezone Plus
Design: https://www.figma.com/file/EIlLPe7KdF4bQXLQVAsDeM/SharezoneApp?type=design&node-id=2372-4484&mode=design&t=sRCMFs02Gv9dau1D-4

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description

I added a new color picker for Sharezone Plus users. Sharezone Plus users first select a base color and can then select an accurate color. The base colors are just all primaries Google Material colors and the accurate colors are all variants of the base color.

## Demo

When a user has Sharezone Plus:

https://github.com/SharezoneApp/sharezone-app/assets/24459435/364ea5d3-8b5e-4a93-ba59-dff2fd91251f

When a user has _not_ Sharezone Plus:

image

## Related tickets

Closes #855
Closes #854

### CODE REVIEW

File: app/lib/groups/src/pages/course/course_edit/design/src/dialog/select_design_dialog.dart [hunk 1]
```diff
@@ -116,19 +140,212 @@ class _Colors extends StatelessWidget {
...
+          opacity: 0,
+          child: IgnorePointer(
+            child: Stack(
+              children: [
+                _PlusBaseColors(),
+                _PlusAccurateColors(baseColor: Colors.amber)
+              ],
+            ),
+          ),
+        ),
```
Comment (nilsreichardt) [inline app/lib/groups/src/pages/course/course_edit/design/src/dialog/select_design_dialog.dart:183]: I'm open for better solutions.

File: app/test_goldens/groups/src/pages/course/course_edit/design/src/dialog/select_design_dialog_test.dart [hunk 2]
```diff
@@ -9,34 +9,121 @@
...
+
+      testGoldens('displays select base color dialog as expected (dark mode)',
+          (tester) async {
+        await pumpSelectDesignDialog(tester, theme: darkTheme);
+
+        await multiScreenGolden(tester, 'select_base_color_dialog_dark');
+      });
+
+      testGoldens(
+          'displays select an accurate color dialog as expected (light mode)',
```
Comment (Jonas-Sander) [inline app/test_goldens/groups/src/pages/course/course_edit/design/src/dialog/select_design_dialog_test.dart:83]:
```suggestion
          'displays select a color shade dialog as expected (light mode)',
```

I think this is better than "accurate color"

**Review by Jonas-Sander — APPROVED** [review]
I think it's surprising for the user that a second dialog with a shade selection will come. I might imagine users being surprised after buying Plus and then not having more colors right on the first screen. Just adding "Grundton auswählen" to the top of the first dialog might already help, because then there is an expectation for a second shade selection.

**Comment by nilsreichardt** [comment]
image

I added "Grundfarbe auswählen" for the first dialog ✅

**Pull request merged**